godo.go

```diff
@@ -223,7 +223,9 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 		return nil, err
 	}
 
-	defer resp.Body.Close()
+	defer func() {
+		err = resp.Body.Close()
```

Comment:
If the `err` is already set, doing this in the defer block will overwrite it and you will lose the earlier, more important error.